### ISSUE TITLE
Exceptions thrown when using BulkSaveChanged - SqlException: Incorrect syntax near ')' and "Parameter count mismatch.",

### ISSUE THREAD

**Issue by danielkaczmarek** [borisdj/EFCore.BulkExtensions#680]
Hi,

Basically Im trying to add operator and units entities at the same save. SaveChanges works fine however BulkSaveChanges throws this SqlException: Incorrect syntax near ')

Please see the attached project
[WpfApp1.zip](https://github.com/borisdj/EFCore.BulkExtensions/files/7660451/WpfApp1.zip)

The other issue I dont know how to reproduce but it gives me "Parameter count mismatch.", here is the call stack

" at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)\r\n at EFCore.BulkExtensions.Helpers.GenericsHelpers.GetPropertiesDefaultValue[T](T value, Type type)\r\n at System.Linq.Enumerable.SelectManySingleSelectorIterator2.ToList()\r\n at System.Linq.Enumerable.ToList[TSource](IEnumerable1 source)\r\n at EFCore.BulkExtensions.Helpers.GenericsHelpers.GetPropertiesWithDefaultValue[T](IEnumerable1 values, Type type)\r\n at EFCore.BulkExtensions.SQLAdapters.SQLServer.SqlOperationsServerAdapter.<MergeAsync>d__51.MoveNext()\r\n at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n at EFCore.BulkExtensions.SQLAdapters.SQLServer.SqlOperationsServerAdapter.d__41.MoveNext()\r\n at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n at EFCore.BulkExtensions.SqlBulkOperation.<MergeAsync>d__51.MoveNext()\r\n at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n at EFCore.BulkExtensions.DbContextBulkTransaction.d__11.MoveNext()\r\n at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n at EFCore.BulkExtensions.DbContextBulkTransactionSaveChanges.<InvokeBulkMethod>d__3.MoveNext()\r\n at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n at EFCore.BulkExtensions.DbContextBulkTransactionSaveChanges.<SaveChangesAsync>d__2.MoveNext()\r\n at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n at EFCore.BulkExtensions.DbContextBulkTransactionSaveChanges.<SaveChangesAsync>d__2.MoveNext()\r\n at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n at EFCore.BulkExtensions.DbContextBulkTransactionSaveChanges.<SaveChangesAsync>d__1.MoveNext()\r\n at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n at EFCore.BulkExtensions.DbContextBulkTransaction.<ExecuteAsync>d__11.MoveNext()\r\n at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n at System.Runtime.CompilerServices.TaskAwaiter.GetResult()\r\n at MAUnitOfWork.UnitOfWorkBase.d__3.MoveNext()

Note: these not happen with a sandard SaveChanges.

Regards
Daniel

**Comment by borisdj** [comment]
You should use _AddRange_ instead of _UpdateRange_.

**Comment by danielkaczmarek** [comment]
> You should use _AddRange_ instead of _UpdateRange_.

It does not make any difference I still get the exception :(

**Comment by danielkaczmarek** [comment]
After updating to the newest version I can see that the first error is gone, however the second one is still there.

**Comment by borisdj** [comment]
Write a test if you manage to reproduce it .

**Comment by danielkaczmarek** [comment]
I did manged to reproduce it, however I'm not sure how would I write a test for it. 
I modified my sample app, so it's very very basic now.

The issue is cause by this property 

    [NotMapped]
    public string this[string pPropertyName] => string.Empty;

The exception message is this: TargetParameterCountException: Parameter count mismatch.

[WpfApp1.zip](https://github.com/borisdj/EFCore.BulkExtensions/files/7692146/WpfApp1.zip)
